// @luminationdev/leadmelabs-plugin-javascript/dist/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  LeadMeLabsConnection: () => LeadMeLabsConnection_default
});
module.exports = __toCommonJS(src_exports);

// src/LeadMeLabsConnection.ts
var net = require("net");
var LeadMeLabsConnection = class {
  logCallback = () => {
  };
  pauseCallback = () => {
  };
  resumeCallback = () => {
  };
  shutdownCallback = () => {
  };
  detailsCallback = () => {
  };
  actionCallback = (message) => {
  };
  server;
  client;
  connect() {
    var PIPE_NAME = "leadme_api";
    var PIPE_PATH = "\\\\.\\pipe\\" + PIPE_NAME;
    this.server = net.createServer((stream) => {
      stream.on("data", (c) => {
        this.handleMessage(c.toString().replace(/\u0000/g, ""));
      });
    }).on("error", () => {
      this.server = null;
    });
    if (this.server) {
      this.server.listen(PIPE_PATH);
    }
  }
  disconnect() {
    this.server.close();
  }
  send(message) {
    var PIPE_NAME = "leadme_parent_api";
    var PIPE_PATH = "\\\\.\\pipe\\" + PIPE_NAME;
    if (!this.client) {
      this.client = net.connect(PIPE_PATH).on("error", () => {
        this.client = null;
      });
    }
    if (this.client) {
      this.client.write(Buffer.from(message, "utf16le"));
    }
  }
  setLogCallback(callback) {
    this.logCallback = callback;
  }
  setPauseCallback(callback) {
    this.pauseCallback = callback;
  }
  setResumeCallback(callback) {
    this.resumeCallback = callback;
  }
  setShutdownCallback(callback) {
    this.shutdownCallback = callback;
  }
  setDetailsCallback(callback) {
    this.detailsCallback = callback;
  }
  setActionCallback(callback) {
    this.actionCallback = callback;
  }
  handleMessage(message) {
    switch (message) {
      case "log":
        this.logCallback();
        break;
      case "pause":
        this.pauseCallback();
        break;
      case "resume":
        this.resumeCallback();
        break;
      case "shutdown":
        this.shutdownCallback();
        break;
      case "details":
        this.detailsCallback();
        break;
      case "action":
      default:
        this.actionCallback(message);
        break;
    }
  }
};
var LeadMeLabsConnection_default = LeadMeLabsConnection;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LeadMeLabsConnection
});
